router/h2/src/main/scala/io/buoyant/router/h2/StreamStatsFilter.scala

```diff
@@ -41,46 +44,118 @@ class StreamStatsFilter(statsReceiver: StatsReceiver)
         case Throw(_) => failures.incr()
       }
     }
+
   }
 
-  //   total number of requests received
+  class FrameStats(
+    protected val stats: StatsReceiver,
+    protected val streamStats: StreamStats
+  ) {
+
+    private[this] val frameBytes = stats.stat("data_frame", "total_bytes")
+
+    def this(stats: StatsReceiver) = this(stats, new StreamStats(stats))
+
+    def apply(
+      startT: Stopwatch.Elapsed,
+      classifyFrame: Try[Option[Frame]] => Unit = { _ => }
+    )(underlying: Stream): Stream = {
+      // TODO: add a fold to `StreamProxy`, so we don't have to use a `var`?
+      var streamFrameBytes: Int = 0
+      // partially evaluate this w/ a reference to the start time
+      val streamStatsT = streamStats(startT)(_)
+      val stream =
+        if (underlying.isEmpty) {
+          // it would be easier to just put this in `stream.onEnd` and handle
+          // both the empty-stream case and the end of a non-empty stream case
+          // in the same block of code, but `join()`ing `rsp.stream.onEnd` with
+          // the `req` stream's `.onEnd` future in `StreamStatsFilter.apply()`
+          // somehow clobbers this callback, so it never triggers.
+          //
+          // i strongly suspect this is a bug in `Stream.onEnd`, since this
```

Comment:
Any progress on this?  I'd like to remove this (and related) comment if it's not a real thing.  Or fix it if it is.